tests/ops/op_math/test_sum.py

```diff
@@ -107,109 +114,154 @@ def compare_and_expand_mat(mat1, mat2):
+    def test_ndim_params_raises_error(self, sum_method):
+        """Test that calling ndim_params raises a ValueError."""
+        sum_op = Sum(qml.PauliX(0), qml.Identity(1))
 
         with pytest.raises(
             ValueError,
             match="Dimension of parameters is not currently implemented for Sum operators.",
         ):
-            sum_op.ndim_params()
+            _ = sum_op.ndim_params
 
-    def test_batch_size_raises_error(self):
-        sum_op = op_sum(qml.PauliX(0), qml.Identity(1))
+    def test_batch_size_raises_error(self, sum_method):
+        """Test that calling batch_size raises a ValueError."""
+        sum_op = Sum(qml.PauliX(0), qml.Identity(1))
 
         with pytest.raises(ValueError, match="Batch size is not defined for Sum operators."):
-            sum_op.batch_size()
+            _ = sum_op.batch_size
 
-    def test_decomposition_raises_error(self):
-        sum_op = op_sum(qml.PauliX(0), qml.Identity(1))
+    def test_decomposition_raises_error(self, sum_method):
+        """Test that calling decomposition() raises a ValueError."""
+        sum_op = Sum(qml.PauliX(0), qml.Identity(1))
 
         with pytest.raises(DecompositionUndefinedError):
             sum_op.decomposition()
 
 
+@pytest.mark.parametrize("sum_method", [sum_using_dunder_method, op_sum])
 class TestMscMethods:
+    """Test dunder methods."""
+
     @pytest.mark.parametrize("ops_lst, ops_rep", tuple((i, j) for i, j in zip(ops, ops_rep)))
-    def test_repr(self, ops_lst, ops_rep):
-        sum_op = Sum(*ops_lst)
-        assert ops_rep == sum_op.__repr__()
+    def test_repr(self, ops_lst, ops_rep, sum_method):
+        """Test __repr__ method."""
+        sum_op = sum_method(*ops_lst)
+        if isinstance(sum_op, Sum):
+            assert ops_rep == repr(sum_op)
 
     @pytest.mark.parametrize("ops_lst", ops)
-    def test_copy(self, ops_lst):
-        sum_op = Sum(*ops_lst)
-        copied_op = copy(sum_op)
+    def test_copy(self, ops_lst, sum_method):
+        """Test __copy__ method."""
+        sum_op = sum_method(*ops_lst)
+        if isinstance(sum_op, Sum):
+            copied_op = copy(sum_op)
 
-        assert sum_op.id == copied_op.id
-        assert sum_op.data == copied_op.data
-        assert sum_op.wires == copied_op.wires
+            assert sum_op.id == copied_op.id
+            assert sum_op.data == copied_op.data
+            assert sum_op.wires == copied_op.wires
 
-        for s1, s2 in zip(sum_op.summands, copied_op.summands):
-            assert s1.name == s2.name
-            assert s1.wires == s2.wires
-            assert s1.data == s2.data
+            for s1, s2 in zip(sum_op.summands, copied_op.summands):
+                assert s1.name == s2.name
+                assert s1.wires == s2.wires
+                assert s1.data == s2.data
 
 
+@pytest.mark.parametrize("sum_method", [sum_using_dunder_method, op_sum])
 class TestMatrix:
+    """Test matrix-related methods."""
```

Comment:
In this case, we want to test explicitly the `op.matrix()` method. While `qml.matrix(op)` does default to `op.matrix()` in most cases. It may be the case that we change this function at some point in the future, so to prevent those bugs from showing up here I think we should swap to `op.matrix()` for all matrix calls, at least in this testing class `TestMatrix`.

Reply:
Swapped to `op.matrix()`